introducing Hauler zero

EndrosG opened the pull request:
I added my Hauler Zero as `Hauler Z` to the standard data files.
Reference: #2396 

I gave it some lore which should be very compatible with the existing lore of the Hauler: Type F series by reactivating that ship class already with the Southern Defense Pact, i.e. before the Free Worlds formed. This adds some background to the Type F series.
The definition is straight forward. The only modification to the regular Hauler outfittings is that I added a liquid nitrogen cooler (+20 tons, all Hauler ships, Type C and Type F, have 22 tons free) because the smaller ship size suffers some heat problems which I did not want to address on the dissipation side.

*It has its own USP*
It delivers firepower at the cost of speed. It lacks some energy, as all Hauler ships do. Since I did not change the stock configuration (despite cooling), it has the best turning of all human light warships which helps to fire its meteor missiles towards the target.

*Balancing*
The fully equipped ship costs 2.084m, fitting perfectly into the set of Quicksilver and Rainmaker.
The 7 free bunks match to the Rainmaker and the Raven, and 400 fuel is the lower end of light warships. Shield charging fits to light warships as well.

*Appendix*
A PR for the 2x file has been made. Both sprites, 1x and 2x, have been created by cutting the C module from the "hauler i" sprite, no real modifications, so I don't even have a .xcf file etc.

jafdy commented:
Umm, but why?

Wrzlprnft commented:
haulers never looked to me as beign modular, more like a logical design choice. i kind of agree with jafdy

jafdy commented:
What I mean to say is; What will this add to the game?

EndrosG commented:
@Wrzlprnft : It is modular lore-wise. The lore of the Hauler series says: a cockpit and an engine block bolted together with (1-3) cargo containers between them. Nobody says you cant connect the cockpit and the engine block directly without any cargo modules. Putting different modules together to a final ship design is part of the work of ship designers...
It is not modular in terms of "modifying ships through outfits after buying them".

@jafdy : Southbound shipyards does not offer any light warship.
They have the Fury Interceptor (755k cr), which is available at Southbound shipyards, Tarazed and various pirate worlds.
They do not have any light warship. Tarazed does not have any light warship either. The pirates of Men, Shaula, Gienah and Algenib do have the Modified Argosy (3.7 mill cr, 6.7k hp). These are currently the only available light warships south from Earth.
Before the introduction of Hauler Type F, the Bastian (5.2 mill cr, 10.9k hp) was the only medium warship at Southbound shipyards, and the Osprey was the only available at Tarazed (8.7 mill cr, 8.8k hp). The pirates of Shaula and Algenib have the Firebird (5.2 mill cr, 9.2k hp).

The Hauler Z(ero) light warship (2.1 mill cr, 4.3k hp) closes a gap in the entire southern region (the dirt belt, the rim and the south). It tells a nice story. It's desing is simple, rocket solid and it can be seen as precursor for the Hauler type F series.

jafdy commented:
Your telling me why it fills a hole in the pattern, not how it improves the game in any way. 
The light warship patch is filled by the Argosy when you add more firepower. With the change to the way raiders work you will now see them reacting to an up-gunned Argosy the same way they would a small warship.

Wrzlprnft commented:
further, unlike the description of the spindle/spire for example, it doesn't say the number of cargo pods changes. like, one time built as hauler 3, it stays a hauler 3, except it gets reworked, but not through unloading. the haulers share parts the same way cars do, at least is thata what i got. so for the sake of transporting the cockpit and the engine habing a hauler Z ingame doesn't make much sense to me, further, a use without a module seems just pretty weird to me. i am not convinced that it is necessary, so when i hit the above mentioned logical border in my mind, i am not convinced. i am open to be convinced, though, and maybe i understood something wrong

MessyMix commented:
I agree that there is no gameplay reason to have a Hauler Z, but then again, I don't see why everyone is _against_ having it.

MessyMix commented on data/ships.txt:
```diff
@@ -1642,6 +1642,62 @@ ship "Hauler III"
+	explode "small explosion" 45
+	explode "medium explosion" 10
+	"final explode" "final explosion small"
+	description "The Southbound Hauler Z is a light warship build from standardized Hauler components."
```
*built

Also, I think "standardized" should be "standard". It seems more accurate to say standard, as the parts were never standardized; they were standard upon inception.

MessyMix commented on data/ships.txt:
```diff
@@ -1642,6 +1642,62 @@ ship "Hauler III"
+	explode "medium explosion" 10
+	"final explode" "final explosion small"
+	description "The Southbound Hauler Z is a light warship build from standardized Hauler components."
+	description "	Centuries ago, various Hauler models made up nearly half of the merchant vessels in service, but today the design is considered somewhat outdated. The first Alpha war led to a temporary high demand for warships. One of the most simple ideas was to build Haulers without any cargo modules. Numerous of these ships were build, but after the Alpha war ended, most of these ships were con ...
```
Don't plagiarize the existing text if it isn't already plagiarized elsewhere.

Also, "Alpha war" should be capitalized (on the "war").

The second time you say "Alpha war", it should be "Alpha Wars".

MessyMix commented on data/ships.txt:
```diff
@@ -1642,6 +1642,62 @@ ship "Hauler III"
+	"final explode" "final explosion small"
+	description "The Southbound Hauler Z is a light warship build from standardized Hauler components."
+	description "	Centuries ago, various Hauler models made up nearly half of the merchant vessels in service, but today the design is considered somewhat outdated. The first Alpha war led to a temporary high demand for warships. One of the most simple ideas was to build Haulers without any cargo modules. Numerous of these ships were build, but after the Alpha war ended, most of these ships were con ...
+	description "	Recently, the old Hauler Z series gained popularity again. As the pirate fleets in the South got stronger every year and the Southern Defense Pact was formed which lead once again to a sudden demand for lightweight warships. This ship still delivers high firepower for a low price."
```
"Recently, demand for the Hauler Z has increased in the South. The ship's return to prominence was fueled in part by the militia's need for a dedicated light warship to suppress the growing pirate presence in the sector. This ship delivers good firepower at an affordable price, which makes it a great choice for captains who need cheap yet robust ships to fend off attacks.

EndrosG replied:
I like your changes, thank you. But I miss one thing. Not sure if that is covered by your second sentense:
The Hauler Z is also a good ship for planetary defense.

EndrosG commented:
Okay, originally I did not intend to copy+paste the lore from the definition file into the PR itself.
[cut out, because MessyMix already suggested changes and there the lore is shown]

**The ships purpose is to serve as warship, not as freighter.** In times of war, uncertainty, etc.
It was (upon my lore) first used during the Alpha War - in the Dirt Belt, in the South and in the Rim. It was a quickly available warship because Southbound shipyards did not had to re-invent anything but just could re-use existing production lines. They simply could temporarily stop assembly of Hauler freighters and assemble Hauler Zero warships with the existing parts.

The Souther Defense Pact needed some warships too, protecting colonies and freighters. Especially after the user encountered that many pirates in the Pact Recon missions.

The Hauler Z's military unique selling proposition is, that it delivers 4 turret mounts (and 2 gun ports) at 2.1m cr. "ready to use" cost.
The Hauler Z delivers a solid anti-missile power, making it a good defender. As a defending ship, it does not need to be fast, but fast turning is good. And it has the fastest turning of all light warships.
Ships used for defense, dont need hundreds of tons of cargo space. They need to be maneuverable and cost efficient.

Lore-wise, the Free Worlds demand for Carriers could be a follow-up upon the reactivation of the Hauler Z model. Asking for more military power instead of ample cargo space.

-----

@Wrzlprnft : A Hauler Z stays a Hauler Z "forever". Except for the "rework" you mentioned which I have in the lore as `but after the Alpha war ended, most of these ships were converted into regular Hauler freighters.`
By using converting instead of refitting I wanted to make clear that this is not a simple outfit thing, but something what only an original construction shipyard can do, having the full assembly line.

If the lore is misleading in that point, feel free to propose a better formulation. I am not an native english speaker, but tried my best.

*I will incorporate some changes as you suggested.*

EndrosG commented:
@MessyMix : I am aware of the fleet testing / modifying task.
My 2nd pilot, having too many interceptors right now (mostly captured during Pact Recon [0]), just accepted Pact Recon [1].

gunqqer commented:
I see nothing wrong with this being added and lore wise everything makes since. Not everything needs perfect gameplay reason nor a lore reason. A mix of both or even just one is fine if it doesn't impede the other.

EndrosG commented:
I think I need to change the lore.
I thought - my error - the Republic existed before the first Alpha War.
Because the description of the Hauler I says, they were introduced in the early days of the Republic (2730+), they did not exist in the first Alpha War (2450) yet.
So the earliest would be around 2720, when some system of the Rim and parts of the Dirt Belt became part of the Republic.
These ships could have been a backbone of the first defense fleets at that time.

Bladewood commented on data/ships.txt:
```diff
@@ -1642,6 +1642,62 @@ ship "Hauler III"
+	"final explode" "final explosion small"
+	description "The Southbound Hauler Z is a light warship which uses standard Hauler components."
+	description "	Centuries ago, various Hauler models made up nearly half of the merchant vessels in service, but today the design is considered somewhat outdated. The first Alpha War led to a temporary high demand for warships. One of the most simple ideas was to build Haulers without any cargo modules. Numerous of these ships were build, but after the Alpha Wars ended, most of these ships were co ...
+	description "	Recently, demand for the Hauler Z has increased in the South. The ship's return to prominence was fueled in part by the militia's need for a dedicated light warship to suppress the growing pirate presence in the sector. This ship delivers good firepower at an affordable price, which makes it a great choice for captains who need cheap yet robust ships to fend off attacks."
```
description seems too long.

MessyMix commented:
@EndrosG: The new description could afford to be a little longer. I also think that it is _necessary_ that you include the lore about the Alpha Wars and the pirates, since it gives a lore reason why the ship came about, and why it stays around right now. The last line is also unnecessary, and it feels like trivia. How about this? (for reference: boxwing description 134 words, my description 137 words.)

>The Hauler Z is the smallest conceivable ship to still retain some defining elements of a Hauler. The namesake of this unique ship stems from the fact that it has zero cargo space. The Hauler Z was introduced in response to the high demand for such a ship during the Alpha Wars. The ship recently returned to prominence, fueled in part by the militia's need for a dedicated light warship to suppress the growing pirate presence in the South.
Because of its impressive weaponry and lack of cargo space, the Hauler Z is a cost-effective ship that delivers good firepower at an affordable price. Merchant captains choose the Hauler Z to act as a cheap yet robust defensive platform to fend off attacks, but it is also sometimes used by pirates to stand up to the milita.

EndrosG commented:
The conflict which I see is the description of the original Hauler (one):
> Several centuries ago, in the early days of the Republic, the sudden increase in commerce and space travel led to high demand for cargo ships. Haulers were first created simply by bolting a cockpit and engine block onto a cargo container, and the basic design has not changed much since then.

That refers to the early 28th century, according to the [historical timeline](https://github.com/endless-sky/endless-sky/wiki/HistoricalTimeline) . The Alphas are mentioned in the 2450's and the 2890's.
I dont really rely on the Alphas. Most important, the Alpha War in the 25th century was won with high tech ships from the Deep.
My new approach was to have them as backbone during the formation of the Republic:
- 2720: Several independent planets in the Core and Rim begin discussing joining together and forming a representative government with Earth as its capital.
- 2730: The Republic is formed, initially comprising Earth, the Rim systems closest to Quarg space, and several Core systems close to Earth, plus the few Dirt Belt worlds that are already inhabited.
- 2740: The Republic imposes trade embargoes on the Deep, the Syndicate, and the paradise worlds, exerting pressure on them to join the Republic.

Alternatively, but may be not that "sweet" would be an introduction in the 2830's. Not that sweet because it feels "very late" with respect to the development since 2780.
- 2830: Freelance captains have begun forming unions and mutual defense pacts, further reducing the amount of piracy in the galaxy and also reducing the power of large corporations.

The latter would perfectly fit to the revival story with the Southern Defence Pact.

EndrosG commented:
@MessyMix Hmm, may be you misunderstood my previous change: I did not copy the 3rd paragraph but only mentioned it as "to be there". So the intention was:
> The Hauler Z is the smallest thinkable Hauler ship. It was introduced just two years after the Hauler freighter. Its name was derived from the fact that this light warship has zero cargo space.
> In the following decades, the Hauler Z formed the backbone of the first military forces of the new Republic. They were also used to enforce the trade embargos in the 2740's.
> Recently, demand for the Hauler Z has increased in the South. The ship's return to prominence was fueled in part by the militia's need for a dedicated light warship to suppress the growing pirate presence in the sector. This ship delivers good firepower at an affordable price. Due to its low speed, it is mostly used for defense purposes.

MessyMix commented:
Great. I _did_ misunderstand your post, although I really liked the Alpha lore. You're right, it's not believable given the timeline.

Okay, here's my revised version of yours. I don't like the middle paragraph, where you mention that the ship "formed the backbone" of the Republic Navy, because I don't think it works like that (I would assume that the role would be taken by Firebirds). I also don't think the following line is necessary, because that seems trivial.

>The Hauler Z is the smallest thinkable Hauler. Introduced only two years after the Hauler freighter, the ship's name derives from the fact that this warship has zero cargo space. In the decades following its release, the Hauler Z saw widespread service in the  of the first Republic military. This ship has recently returned to prominence in the South, fueled in part by the militia's need for a dedicated light warship to suppress the growing pirate presence in the sector.

>Because of its great combat ability, the Hauler Z is extremely cost effective, delivering an inordinate amount of firepower for an affordable price. Merchant captains utilize the Hauler Z as a cheap and robust defensive platform to fend off attacks, although it has seen use in piracy as a ship that can stand up to militia fleets.

Amazinite commented on data/ships.txt:
```diff
@@ -1642,6 +1642,62 @@ ship "Hauler III"
+			"blast radius" 40
+			"shield damage" 400
+			"hull damage" 200
+			"hit force" 800
```
Should be 600

EndrosG commented:
@Amazinite : fixed, thanks
@MessyMix : I did not name the "Navy". When the Republic was formed, they did not yet have a Navy. As the Navy was formed, "real" warships were designed and replaced the "temporary solution". I accepted your new wording and really want to thank you for your assistance. :+1: 
I fixed your typo in that sentense (in the of the first ...) and still did not fully agree to your last sentense. I modified it limiting the usage to small convoys, pointing to the Bastion as successor, and skipping the pirate usage. It is (imho) obvious, that they use captured ships for their own purposes, but they still prefer the Modified Argosy.

endless-sky commented:
I don't think this is distinct enough from other human ships to be worth adding. Humans already have a huge variety of ships compared to most other species, so adding any more human ships is a very low priority.

Elyssaen commented:
You should talk to DerpyHorse about getting it in [Shields Up](https://steamcommunity.com/app/404410/tradingforum/1843493219420821718/?l=czech). It's one of the most popular plug-ins, and incorporates Phil's Flavours, which add a _lot_ of flavourful, good-looking human ships. Your two ships (the Hauler Z and Large Star Barge, though I'd prefer the word 'Heavy' there) would fit right in, they'd just need to be added to fleet definitions.

First off, the Hauler Z is surprisingly good-looking. Never thought I'd find a Hauler attractive in any form.

Mechanically, it seems pretty balanced, and does fill a niche. That said, it's a little unusual. Compared to the Raven, it's only a little less costly, and is a trade-off of speed and toughness (the Raven) against the Hauler Z's potential firepower. I think the upshot of this kind of balance is the Hauler Z is unlikely to be a _player's_ ship, which is a shame.

The low shielding can be justified, but what really bugs me is that there's _no_ cargo space. I get that that's the concept, but it means you really can't entertain the thought of using it as a player ship, at all, without a Cargo Expansion. Could it instead be the case that the name 'Z' was used in place of 'Zero' because, some historical upgrade to it decided to fit a small 10 ton cargo bay to give it some additional versatility? Just say it's a pod underneath, add 3 tons of mass if you must.

EndrosG commented:
That's at least worth additional consideration.
I always *accepted* the `Hauler Z` to not have any cargo space - and it always can be added with the cargo extension, as you mentioned - because it was not designed as a player ship in the first place.

`Heavy Star Barge` sounds like "this is a heavy freighter" then, doesn't it?

Elyssaen commented:
There's a Heavy Shuttle. I s'pose there it's a transport, but even so I don't think the 'Heavy' prefix would make either sound like a heavy freighter.

And yeah, I guess since it seems more like an NPC ship as it is, perhaps the cargo space doesn't matter as much. Up to you.

EndrosG commented:
And would you actually prefer `Heavy Star Barge` or just `Heavy Barge`?

Elyssaen commented:
Probably the former, myself.
